optuna/trial/_frozen.py

```diff
@@ -166,29 +170,72 @@ def _validate(self):
                 )
 
     @property
-    def distributions(self):
-        # type: () -> Dict[str, BaseDistribution]
+    def number(self) -> int:
+        return self._number
```

Comment:
These all getters and setters should be placed after public functions (`set_system_attr`). The function order should match with that of `optuna/trial/_trial.py`.